using libKTX on Linux when linking report errors on Threads::Threads

I have tried to use the RPM 4.3-beta modified from this PR #830 in a project.

when compiliing it is reporting that cannot find `Threads::Threads`


I have tried to change the `PUBLIC` here:
https://github.com/KhronosGroup/KTX-Software/blob/main/CMakeLists.txt#L649

to `PRIVATE`

and that issue seems resolved.

reverted the chane in #830 .

i am linking to shared: using `KTX::ktx` in `target_link_libraries`.:

```
target_link_libraries(${PROJECT_NAME} PUBLIC
    KTX::ktx
)
```

The error is:
```
[cmake] CMake Error at /usr/lib/cmake/ktx/KtxTargets.cmake:56 (set_target_properties):
[cmake]   The link interface of target "KTX::ktx" contains:
[cmake] 
[cmake]     Threads::Threads
[cmake] 
[cmake]   but the target was not found.  Possible reasons include:
[cmake] 
[cmake]     * There is a typo in the target name.
[cmake]     * A find_package call is missing for an IMPORTED target.
[cmake]     * An ALIAS target is missing.
[cmake] 
[cmake] Call Stack (most recent call first):
[cmake]   /usr/lib/cmake/ktx/KtxConfig.cmake:7 (include)
[cmake]   CMakeLists.txt:24 (find_package)
[cmake] 
```

The file `KtxTargets.cmake` at line 56 has this statement:
```
set_target_properties(KTX::ktx PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "\$<\$<CONFIG:Debug>:_DEBUG;DEBUG>;KTX_FEATURE_KTX1;KTX_FEATURE_KTX2;KTX_FEATURE_WRITE"
  INTERFACE_COMPILE_FEATURES "c_std_99;cxx_std_11"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "dl;Threads::Threads"
)
```



if that would be the case with static, maybe a cmake "`if(static) private else public`" could do the trick eventually...

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -646,7 +646,7 @@ macro(common_libktx_settings target enable_write library_type)
         find_package(Threads REQUIRED)
         target_link_libraries(
             ${target}
-        PUBLIC
+            PRIVATE
             dl
             Threads::Threads
         )

